nixos/modules/system/activation/top-level.nix

```diff
@@ -83,6 +82,7 @@ let
       done
 
       mkdir $out/bin
+      export localeArchive=${lib.optionalString (pkgs.stdenv.isLinux or false) "${pkgs.glibcLocales}/lib/locale/locale-archive"}
```

Comment:
This should use `"${config.i18n.glibcLocales}/lib/locale/locale-archive"` so the NixOS locale configuration options are respected.

And no need to make conditional for Linux.

Reply:
Done.